src/System.IO.FileSystem.AccessControl/tests/DirectoryObjectSecurityTests.cs

```diff
@@ -0,0 +1,232 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+using System.DirectoryServices;
+using System.Security.Principal;
+using Xunit;
+
+namespace System.Security.AccessControl
+{
+    public class DirectoryObjectSecurityTests
+    {
+        [Fact]
+        public void ObjectInitialization_DefaultConstructor_Success()
+        {
+            var activeDirectorySecurity = new ActiveDirectorySecurity();
+            Assert.NotNull(activeDirectorySecurity);
+        }
+
+        [Fact]
+        public void ObjectInitialization_CommonSecurityDescriptor_Success()
+        {
+            
+            CommonSecurityDescriptor descriptor = new CommonSecurityDescriptor(false, false, string.Empty);
+            var customObjectSecurity = new CustomDirectoryObjectSecurity(descriptor);
+            Assert.NotNull(customObjectSecurity);
+        }
+
+        [Fact]
+        public void ObjectInitialization_InvalidArgument()
+        {
+            Assert.Throws<ArgumentNullException>(() => new CustomDirectoryObjectSecurity(null));
+        }
+
+        [Fact]
+        public void GetAccessRules_InvalidTargetType()
+        {
+            var activeDirectorySecurity = new ActiveDirectorySecurity();
+            Assert.Throws<ArgumentException>(() => 
+               activeDirectorySecurity
+               .GetAccessRules(false, false, typeof(System.Security.Principal.GenericPrincipal)));
+        }
+
+        [Fact]
+        public void GetAccessRules_TargetType_SecurityIdentifier_ReturnsValidObject()
+        {
+            CommonSecurityDescriptor descriptor = new CommonSecurityDescriptor(false, false, string.Empty);
+            var customObjectSecurity = new CustomDirectoryObjectSecurity(descriptor);
+
+            AuthorizationRuleCollection ruleCollection =
+                customObjectSecurity
+               .GetAccessRules(false, false, typeof(System.Security.Principal.SecurityIdentifier));
+
+            Assert.NotNull(ruleCollection);
```

Comment:
Is there anything more we can validate about the rule collection beyond GetAccessRules returning a non-null instance?  Same for other occurrences of this.